### PR TITLE
Fakewebapi simulare a Fleet API endpoint for testing the checkin

### PR DESCRIPTION
**Pull request by ph**
This PR creates a crude fake webserver to allow a beat agent to enroll
and receives action from a remote host.

@michalpristas This is really a quick and dirty webserver just to simulate a subset of the Fleet API, There are currently no validation for the action that are acked.

### CODE REVIEW

File: x-pack/agent/dev-tools/cmd/fakewebapi/fetch.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+APIKEY=${1:-"abc123"}
+AGENTID=${2:-"agent007"}
```
Comment (michalpristas) [inline x-pack/agent/dev-tools/cmd/fakewebapi/fetch.sh]: 😄

File: x-pack/agent/dev-tools/cmd/fakewebapi/main.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	w.Write(b)
+	log.Println("Enroll response:", string(b))
+}
+
+func handlerCheckin(w http.ResponseWriter, r *http.Request) {
+	mutex.Lock()
```
Comment (michalpristas) [inline x-pack/agent/dev-tools/cmd/fakewebapi/main.go]: what this lock locks?
you are responding with same response always, what about preparing mashalled response upfront

Comment (ph) [reply x-pack/agent/dev-tools/cmd/fakewebapi/main.go]: Well, I wanted to make it more freeform, you can actually push a new configuration using the `handlerAction`

File: x-pack/agent/dev-tools/cmd/fakewebapi/fetch.sh (hunk 1 above)
Comment (ph) [reply x-pack/agent/dev-tools/cmd/fakewebapi/fetch.sh]: shameless influence from a trailer.

**Pull request merged**